### PR TITLE
[G2M] Availability Search

### PR DESCRIPTION
**Pull request by Goondrious**
- basic interface for querying tasks, with the end goal of checking dev-avail project
- restructured project
- added lint configuration from Snoke

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -4,9 +4,27 @@
   "main": "index.js",
   "license": "MIT",
   "scripts": {
-    "routine": "node routine.js"
+    "routine": "node ./src/routine.js",
+    "availability": "env $(print-env) node ./src/availability",
```
Comment (woozyking) [inline package.json]: this won't work without the commented out logic in `availability.js`. see comment there

File: src/availability.js [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+
+/*
+  USAGE
+  const ret = await getTasksForProject({
+    sectionName: 'in office',
+    userName: 'Leo Li',
+    now: false,
+  })
+*/
+module.exports = getTasksForProject
```
Comment (woozyking) [inline src/availability.js]: you can conditionally expose this as exports or runnable depending on execution context:

```js
if (require.main === module) {
  // runnable context ($ node ...)
  // 
} else {
  // library context
  module.exports = getTasksForProject
}
```

File: src/routine.js [hunk 3]
```diff
@@ -1,8 +1,16 @@
 const client = require('./client')
 
+<<<<<<< HEAD:routine.js
```
Comment (woozyking) [inline src/routine.js]: a few unresolved conflicts in this file

File: src/routine.js [hunk 4]
```diff
@@ -1,8 +1,10 @@
 const client = require('./client')
 
+
 const { ASANA_PROJECT = '1152701043959235' } = process.env
 
-const pastFilter = task => new Date(`${task.due_at ? task.due_at : `${task.due_on}T23:59:59.999Z`}`) < new Date()
+const pastFilter = (task) => new Date(`${task.due_at ? task.due_at : `${task.due_on}T23:59:59}`}`)
```
Comment (woozyking) [inline src/routine.js]: i think there was a reason to do `.999Z` to make it fully compliant with timestamp parsing.

File: src/routine.js [hunk 5]
```diff
@@ -1,8 +1,10 @@
 const client = require('./client')
 
+
 const { ASANA_PROJECT = '1152701043959235' } = process.env
 
-const pastFilter = task => new Date(`${task.due_at ? task.due_at : `${task.due_on}T23:59:59.999Z`}`) < new Date()
+const pastFilter = (task) => new Date(`${task.due_at || `${task.due_on}T23:59:59.999Z}`}`)
```
Comment (woozyking) [inline src/routine.js]: need `< new Date()` to test whether it's past

Comment (Goondrious) [reply src/routine.js]: this is on the next line

**Pull request merged**